specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json

```diff
@@ -0,0 +1,1671 @@
+      "get": {
+        "tags": [
+          "FirewallRules"
+        ],
+        "operationId": "FirewallRules_ListByServer",
+        "x-ms-examples": {
+          "FirewallRuleList": {
+            "$ref": "./examples/FirewallRuleListByServer.json"
+          }
+        },
+        "description": "List all the firewall rules in a given PostgreSQL server.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/FirewallRuleListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/singleServers/{serverName}/configurations": {
+      "get": {
+        "tags": [
+          "Configurations"
+        ],
+        "operationId": "Configurations_ListByServer",
+        "x-ms-examples": {
+          "ConfigurationList": {
+            "$ref": "./examples/ConfigurationListByServer.json"
+          }
+        },
+        "description": "List all the configurations in a given server.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/singleServers/{serverName}/updateConfigurations": {
+      "post": {
```

Comment:
>post [](start = 7, length = 4)

Why is configuration update modeled as a POST instead of PUT? Resource modifications via POST are generally an ARM anti-pattern.

Reply:
Agreed and will change it to PUT.

Reply:
Did you mean to make this a PUT on `configurations/{configurationName}`? Or are `updateConfigurations` different than `configurations` resources?

---
In reply to: [451866917](https://github.com/Azure/azure-rest-api-specs/pull/10020#discussion_r451866917) [](ancestors = 451866917)

Reply:
update configurations.

Reply:
I may have misunderstood something here. Is the intent here to make a bulk update of /configurations resources? If yes, can you take a look at "Deep PUT" on the ARM wiki to see if it'll work for you here?

---
In reply to: [451974411](https://github.com/Azure/azure-rest-api-specs/pull/10020#discussion_r451974411) [](ancestors = 451974411)

Reply:
@majastrz we are looking into this options.

Reply:
please dont do POST, use Deep PUT for the bulk scenario. POST will bypass policy, it can by pass RBAC etc. Not a supported pattern in Azure.